raftwal/storage.go

```diff
@@ -85,26 +85,38 @@ type indexRange struct {
 	from, until uint64 // index range for deletion, until index is not deleted.
 }
 
-// Init initializes returns a properly initialized instance of DiskStorage.
-// To gracefully shutdown DiskStorage, store.Closer.SignalAndWait() should be called.
+// Init initializes an instance of DiskStorage without encryption.
 func Init(dir string) *DiskStorage {
+	ds, err := InitEncrypted(dir, nil)
+	x.Check(err)
+	return ds
+}
+
+// InitEncrypted initializes returns a properly initialized instance of DiskStorage.
+// To gracefully shutdown DiskStorage, store.Closer.SignalAndWait() should be called.
+func InitEncrypted(dir string, encKey x.SensitiveByteSlice) (*DiskStorage, error) {
```

Comment:
There is no point in returning the error since we're doing x.Check(err) everywhere after calling `InitEncrypted`.

Reply:
We needed to return the error so that we can test the `InitEncrypted` function for various cases. For example, calling `InitEncrypted(dir, wrongEncryptionKey)` will Fatal and there will be no way to do these tests. Had `x.Check()` been a panic instead of a Fatal, then we could have caught the panic by `require.PanicsWithError()`. Do you have some idea/workaround?